lib/events.js

```diff
@@ -307,3 +345,10 @@ function spliceOne(list, index) {
     list[i] = list[k];
   list.pop();
 }
+
+function arrayClone(arr, len) {
+  var arr_clone = new Array(len);
+  for (var i = 0; i < len; i += 1)
```

Comment:
on jsperf (http://jsperf.com/new-array-vs-splice-vs-slice/19) the fastest methods for array cloning are `.slice()` and `.concat()`